### PR TITLE
fix: cover more error reason messages

### LINKED ISSUE #758: [Feature]: Need to cover more error reason messages.
### Checklist

- [X] I've searched for similar issues and couldn't find anything matching
- [X] I've discussed this feature request in the [K8sGPT Slack](https://k8sgpt.slack.com) and got positive feedback

### Is this feature request related to a problem?

None

### Problem Description

Need to cover more error reasons in [pkg/analyzer/pod.go](https://github.com/k8sgpt-ai/k8sgpt/blob/24132c2d87024157009589cf2bd410bac2a26241/pkg/analyzer/pod.go#L63).

## 8 Error reason messages list to add
1. PreCreateHookError
2. CreateContainerError
3. PreStartHookError
4. RunContainerError
5. ImageInspectError
6. ErrImagePull
7. ErrImageNeverPull
8. InvalidImageName

### Error definitions in kuberuntime_container.go
1-4  : https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kuberuntime/kuberuntime_container.go#L178

### Error definitions in image_manager.go
5-8 : https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/images/image_manager.go#L101

### Solution Description

Add conditions in [pkg/analyzer/pod.go](https://github.com/k8sgpt-ai/k8sgpt/blob/24132c2d87024157009589cf2bd410bac2a26241/pkg/analyzer/pod.go#L63).

### Benefits

More k8s error reason messages can be covered in k8sgpt.

### Potential Drawbacks

none

### Additional Information

If it is okay, I will submit the PR.

### PR DESCRIPTION
**Pull request by jmyung**
Closes #758 

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [x] My code requires changes to the documentation
- [x] I have updated the documentation as required
- [x] All the tests have passed

### CODE REVIEW

File: pkg/analyzer/pod.go [hunk 1]
```diff
@@ -60,7 +60,7 @@ func (PodAnalyzer) Analyze(a common.Analyzer) ([]common.Result, error) {
 		// Check through container status to check for crashes or unready
 		for _, containerStatus := range pod.Status.ContainerStatuses {
 			if containerStatus.State.Waiting != nil {
-				if containerStatus.State.Waiting.Reason == "CrashLoopBackOff" || containerStatus.State.Waiting.Reason == "ImagePullBackOff" || containerStatus.State.Waiting.Reason == "CreateContainerConfigError" {
+				if containerStatus.State.Waiting.Reason == "CrashLoopBackOff" || containerStatus.State.Waiting.Reason == "ImagePullBackOff" || containerStatus.State.Waiting.Reason == "CreateContainerConfigError" || containerStatus.State.Waiting.Reason == "PreCreateHookError" || containerStatus.State.Waiting.Reason == "CreateContainerError" || containerStatus.State.Waiting.Reason == "PreStartHookError" || containerStatus.State.Waiting.Reason == "RunContainerError" || containerStatus.State.Waiting.Reason == "ImageInspectError" || containerStatus.State.Waiting.Reason == "ErrImagePull" || containerStatus.State.Waiting.Reason == "ErrImageNeverPull" || containerStatus.State.Waiting.Reason == "InvalidImageName" {
```
Comment (arbreezy) [inline pkg/analyzer/pod.go:63]: I suggest to make it a variable for readability e.g ```
containerUnhealthyStatus := containerStatus.State.Waiting.Reason == "CrashLoopBackOff" || containerStatus.State.Waiting.Reason == "ImagePullBackOff" || containerStatus.State.Waiting.Reason == "CreateContainerConfigError" || containerStatus.State.Waiting.Reason == "PreCreateHookError" || containerStatus.State.Waiting.Reason == "CreateContainerError" || containerStatus.State.Waiting.Reason == "PreStartHookError" || containerStatus.State.Waiting.Reason == "RunContainerError" || containerStatus.State.Waiting.Reason == "ImageInspectError" || containerStatus.State.Waiting.Reason == "ErrImagePull" || containerStatus.State.Waiting.Reason == "ErrImageNeverPull" || containerStatus.State.Waiting.Reason == "InvalidImageName"```

Comment (jmyung) [reply pkg/analyzer/pod.go]: Thank you. @arbreezy 😄 

For more refactoring, I modified conditions.

**Review by AlexsJones — APPROVED** [review]
Nice ie thanks @jmyung

**Review by arbreezy — APPROVED** [review]

File: pkg/analyzer/pod.go (hunk 1 above)
Comment (arbreezy) [reply pkg/analyzer/pod.go]: Nice good job @jmyung

**Comment by AlexsJones** [comment]
@jmyung thanks for the PR, if you can update the branch we can merge it in

**Pull request merged**